.Lib9c.Tests/Action/RetrieveAvatarAssetsTest.cs

```diff
@@ -0,0 +1,162 @@
+        public void LoadPlainValue()
+        {
+            var avatarAddress = new PrivateKey().Address;
+            var expectedValue = Dictionary.Empty
+                .Add("type_id", RetrieveAvatarAssets.TypeIdentifier)
+                .Add("values", Dictionary.Empty.Add("a", avatarAddress.Serialize()));
+
+            // Let's assume that serializedAction is the serialized representation of the action, which might be obtained through some other part of your tests
+            var action = new RetrieveAvatarAssets();
+            action.LoadPlainValue(expectedValue);
+
+            var plainValue = (Dictionary)action.PlainValue;
+            var values = (Dictionary)plainValue["values"];
+
+            Assert.Equal((Text)RetrieveAvatarAssets.TypeIdentifier, plainValue["type_id"]);
+            Assert.Equal(avatarAddress, values["a"].ToAddress());
+        }
+
+        [Fact]
+        public void Execute()
+        {
+            var agentState = _state.GetAgentState(_signer);
+            Assert.NotNull(agentState);
+            var avatarAddress = agentState.avatarAddresses[0];
+            var prevState = _state.MintAsset(
+                new ActionContext
+            {
+                Signer = _minter,
+            },
+                avatarAddress,
+                1 * _currency
+            );
+            Assert.Equal(1 * _currency, prevState.GetBalance(avatarAddress, _currency));
+
+            var action = new RetrieveAvatarAssets(avatarAddress);
+            var nextState = action.Execute(new ActionContext
+            {
+                PreviousState = prevState,
+                BlockIndex = 1L,
+                Signer = _signer,
+                RandomSeed = 0,
+            });
+            Assert.Equal(0 * _currency, nextState.GetBalance(avatarAddress, _currency));
+            Assert.Equal(1 * _currency, nextState.GetBalance(_signer, _currency));
+        }
+
+        [Fact]
+        public void Execute_Throw_FailedLoadStateException()
+        {
+            var avatarAddress = new PrivateKey().Address;
+            var action = new RetrieveAvatarAssets(avatarAddress);
+
+            var context = new ActionContext()
+            {
+                BlockIndex = 0,
+                PreviousState = new World(new MockWorldState()),
+                RandomSeed = 0,
+                Signer = avatarAddress,
+            };
+
+            Assert.Throws<FailedLoadStateException>(() => action.Execute(context));
+        }
+
+        [Fact]
+        public void Execute_Throw_InsufficientBalanceException()
+        {
+            var agentState = _state.GetAgentState(_signer);
+            Assert.NotNull(agentState);
+            var avatarAddress = agentState.avatarAddresses[0];
+            Assert.Equal(0 * _currency, _state.GetBalance(avatarAddress, _currency));
+
+            var action = new RetrieveAvatarAssets(avatarAddress);
+            Assert.Throws<InsufficientBalanceException>(() => action.Execute(new ActionContext
```

Comment:
```
Expected: typeof(Libplanet.Action.State.InsufficientBalanceException)
Actual:   typeof(System.ArgumentOutOfRangeException): The value to transfer has to be greater than zero. (Parameter 'value')
```

It seems to throw ArgumentOutOfRangeException.

https://docs.libplanet.io/4.0.6/api/Libplanet.Action.State.IAccount.html#Libplanet_Action_State_IAccount_TransferAsset_Libplanet_Action_IActionContext_Libplanet_Crypto_Address_Libplanet_Crypto_Address_Libplanet_Types_Assets_FungibleAssetValue_System_Boolean_

- `arg <= 0` => ArgumentOutOfRangeException
- `arg > 0` => InsufficientBalanceException